### PR TITLE
Removing dnac_log_level from code along with fixing is…

### PR DESCRIPTION
**Pull request by abimishr**
Problem description:  Discovery can't take more than 5 CLI credentials. Along with that, we haven't covered range based discovery.
Fix: Now the customer has the option to add the number of CLI credentials to be added. We have changed the IP address list input method.
Test cases: Tested for CDP, LLDP, MULTI RANGE and SINGLE RANGE
Limitation: Deletion is not tested

### CODE REVIEW

File: plugins/modules/discovery_intent.py [hunk 1]
```diff
@@ -465,36 +465,41 @@ def preprocessing_devices_info(self, devices_list=None):
...
+        elif self.validated_config[0].get('discovery_type') == "RANGE":
+            if len(ip_address_list) == 1:
+                if len(str(ip_address_list[0]).split("-")) == 2:
+                    ip_address_list = ip_address_list[0]
+                else:
+                    ip_address_list = "{0}-{1}".format(ip_address_list[0], ip_address_list[0])
+            else:
+                self.log("IP Address list's length is longer than 1", "ERROR")
+                self.module.fail_json(msg="IP Address list's length is longer than 1", response=[])
         else:
```
Comment (madhansansel) [inline plugins/modules/discovery_intent.py:494]: As many lines are repeated in this API:

    if discovery_type in ["SINGLE", "CDP", "LLDP"]:
        if len(ip_address_list) == 1: 
            processed_ip_addresses = ip_address_list[0] 
        else:
            self.handle_discovery_error()
    elif discovery_type == "CIDR":
        processed_ip_addresses = self.process_discovery_cidr(ip_address_list)
    elif discovery_type == "RANGE":
        processed_ip_addresses = self.process_discovery_range(ip_address_list)
    else:
        processed_ip_addresses = self.process_discovery_other(ip_address_list)

    self.log("Collected IP address/addresses are {0}".format(str(processed_ip_addresses)), "INFO")

And followed by the functions?

```
def handle_discovery_error(self):
    self.log("The length of the IP address list is longer than 1", "ERROR")
    self.module.fail_json(msg="he length of the IP address list is longer than 1", response=[])

def process_cidr(self, ip_address_list):
    if  len(ip_address_list) != 1:
                self.handle_discovery_error()
                return

    prefix_length = self.validated_config[0].get('prefix_length')
#check prefix length if required...
    return "{0}/{1}".format(ip_address_list[0], prefix_length))
```

Likewise you can write other APIs...

File: plugins/modules/discovery_intent.py [hunk 2]
```diff
@@ -465,36 +465,41 @@ def preprocessing_devices_info(self, devices_list=None):
...
                 ip_address_list = str(ip_address_list) + "/" + str(self.validated_config[0].get('prefix_length'))
             else:
-                self.log("Device list's length is longer than 1", "ERROR")
-                self.module.fail_json(msg="Device list's length is longer than 1", response=[])
+                self.log("IP Address list's length is longer than 1", "ERROR")
+                self.module.fail_json(msg="IP Address list's length is longer than 1", response=[])
+        elif self.validated_config[0].get('discovery_type') == "RANGE":
+            if len(ip_address_list) == 1:
+                if len(str(ip_address_list[0]).split("-")) == 2:
+                    ip_address_list = ip_address_list[0]
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/discovery_intent.py:488]: We are actually fetching the data for performing the API call

File: plugins/modules/discovery_intent.py [hunk 3]
```diff
@@ -147,10 +134,13 @@
       snmp_version:
         description: Version of SNMP (v2/v3)
         type: str
-        required: true
       timeout:
         description: Time to wait for device response in seconds
         type: int
+      cli_cred_len:
+       description: Total CLI credentials that needs to be used. This value can vary from 1 to 5.
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/discovery_intent.py:141]: Resolved

File: plugins/modules/discovery_intent.py [hunk 4]
```diff
@@ -441,22 +441,22 @@ def get_devices_list_info(self):
...
-    def preprocessing_devices_info(self, devices_list=None):
+    def preprocessing_devices_info(self, ip_address_list=None):
         """
         Preprocess the devices' information. Extract the IP addresses from
         the list of devices and perform additional processing based on the
         'discovery_type' in the validated configuration.
 
         Parameters:
-          - devices_list: The list of devices to preprocess. If not
+          - ip_address_list: The list of devices to preprocess. If not
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/discovery_intent.py:459]: yeah
It is resolved now

File: plugins/modules/discovery_intent.py [hunk 5]
```diff
@@ -465,36 +465,41 @@ def preprocessing_devices_info(self, devices_list=None):
...
+            ip_address_list = []
 
         self.log("Discovery type passed for the discovery is {0}".format(self.validated_config[0].get('discovery_type')), "INFO")
         if self.validated_config[0].get('discovery_type') in ["SINGLE", "CDP", "LLDP"]:
             if len(ip_address_list) == 1:
                 ip_address_list = ip_address_list[0]
             else:
-                self.log("Device list's length is longer than 1", "ERROR")
-                self.module.fail_json(msg="Device list's length is longer than 1", response=[])
+                self.log("IP Address list's length is longer than 1", "ERROR")
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/discovery_intent.py:476]: Resolved

File: plugins/modules/discovery_intent.py [hunk 6]
```diff
@@ -465,36 +465,41 @@ def preprocessing_devices_info(self, devices_list=None):
...
+                self.module.fail_json(msg="IP Address list's length is longer than 1", response=[])
         elif self.validated_config[0].get('discovery_type') == "CIDR":
             if len(ip_address_list) == 1 and self.validated_config[0].get('prefix_length'):
                 ip_address_list = ip_address_list[0]
                 ip_address_list = str(ip_address_list) + "/" + str(self.validated_config[0].get('prefix_length'))
             else:
-                self.log("Device list's length is longer than 1", "ERROR")
-                self.module.fail_json(msg="Device list's length is longer than 1", response=[])
+                self.log("IP Address list's length is longer than 1", "ERROR")
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/discovery_intent.py:483]: Resolved

File: plugins/modules/discovery_intent.py [hunk 7]
```diff
@@ -465,36 +465,41 @@ def preprocessing_devices_info(self, devices_list=None):
...
+                self.module.fail_json(msg="IP Address list's length is longer than 1", response=[])
+        elif self.validated_config[0].get('discovery_type') == "RANGE":
+            if len(ip_address_list) == 1:
+                if len(str(ip_address_list[0]).split("-")) == 2:
+                    ip_address_list = ip_address_list[0]
+                else:
+                    ip_address_list = "{0}-{1}".format(ip_address_list[0], ip_address_list[0])
+            else:
```
[the remark this answers is not in the document]
Comment (abimishr) [reply plugins/modules/discovery_intent.py:491]: Resolved

**Review by madhansansel — APPROVED** [review]

**Pull request merged**